app/src/main/java/com/ncnf/database/builder/DatabaseObjectBuilder.java

```diff
@@ -0,0 +1,10 @@
+package com.ncnf.database.builder;
+
+import java.util.Map;
+
+public abstract class DatabaseObjectBuilder<T> {
+
+    public abstract T toObject(String uuid, Map<String, Object> data);
```

Comment:
To reduce generics, you could have a common interface that all the "Objects" in the map implement, then going from <T> to <T extends ...> But this is a minor detail